Source/AviationWeather/Source/metar.cpp

```diff
@@ -1395,6 +1400,32 @@ void metar_info::parse()
     detail::parse_altimeter(*this, baseMetar);
 }
 
+cloud_layer metar_info::ceiling() const
+{
+    for (auto it = sky_condition_group.begin(); it != sky_condition_group.end(); ++it)
+    {
+        if (it->sky_cover == sky_cover_type::broken || it->sky_cover == sky_cover_type::overcast)
+        {
+            return *it;
+        }
+    }
+    // check vertical visibility?
```

Comment:
Just checked AC 00-45G (page 3-13 and 3-14), looks like vertical visibility (indefinite ceiling) does count as a ceiling.  Fixed in next commit.